abi.go

```diff
@@ -44,25 +50,42 @@ func newMapABIFromFd(fd *internal.FD) (string, *MapABI, error) {
 
 	return "", &MapABI{
 		MapType(info.mapType),
+		MapID(info.id),
 		info.keySize,
 		info.valueSize,
 		info.maxEntries,
 		info.flags,
+		internal.CString(info.mapName[:]),
+		UnspecifiedProgram,
 	}, nil
 }
 
 func newMapABIFromProc(fd *internal.FD) (*MapABI, error) {
 	var abi MapABI
-	err := scanFdInfo(fd, map[string]interface{}{
+	fields := map[string]interface{}{
 		"map_type":    &abi.Type,
 		"key_size":    &abi.KeySize,
 		"value_size":  &abi.ValueSize,
 		"max_entries": &abi.MaxEntries,
 		"map_flags":   &abi.Flags,
-	})
+		"map_id":      &abi.ID,
+	}
+	if abi.Type == ProgramArray {
+		fields["owner_prog_type"] = &abi.OwnerProgType
```

Comment:
So only ProgramArray will have this populated? Do other map types not honour this field?

Reply:
Exactly, see here https://github.com/torvalds/linux/blob/63623fd44972d1ed2bfb6e0fb631dfcf547fd1e7/kernel/bpf/syscall.c#L542-L568